From aeae03a31d86b9c7304aee90e4a131eac42cf1fa Mon Sep 17 00:00:00 2001
From: "Guoniu.Zhou" <guoniu.zhou@nxp.com>
Date: Tue, 13 Mar 2018 18:03:02 +0800
Subject: [PATCH 3508/5242] MLK-17790-1: dts: add clocks for CI_PI subsystme

commit  ccc0d7ad4a5484c9a2dfb78032affe4a833a69cf from
https://source.codeaurora.org/external/imx/linux-imx.git

Because QXP will drop parent clock info of CI_PI SS
after system suspend/resume, so driver need to record
the relationship of clocks.

Reviewed-by: sandor.yu <sandor.yu@nxp.com>
Signed-off-by: Guoniu.Zhou <guoniu.zhou@nxp.com>
(cherry picked from commit d4b94c2b128850bed8ec4c00ce0c0e3f86a05fcd)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi |    7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi
index 5fba80a..e11235a 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi
@@ -1725,8 +1725,11 @@
 			compatible = "fsl,mxc-parallel-csi";
 			reg = <0x0 0x58261000 0x0 0x1000>;
 			clocks = <&clk IMX8QXP_PARALLEL_CSI_PIXEL_CLK>,
-				   <&clk IMX8QXP_PARALLEL_CSI_IPG_CLK>;
-			clock-names = "pixel", "ipg";
+				   <&clk IMX8QXP_PARALLEL_CSI_IPG_CLK>,
+				   <&clk IMX8QXP_PARALLEL_CSI_CLK_SEL>,
+				   <&clk IMX8QXP_PARALLEL_CSI_PER_CLK_DIV>,
+				   <&clk IMX8QXP_PARALLEL_CSI_CLK_DPLL>;
+			clock-names = "pixel", "ipg", "sel", "div", "dpll";
 			assigned-clocks = <&clk IMX8QXP_PARALLEL_CSI_CLK_SEL>,
 							<&clk IMX8QXP_PARALLEL_CSI_PER_CLK_DIV>;
 			assigned-clock-parents = <&clk IMX8QXP_PARALLEL_CSI_CLK_DPLL>;
-- 
1.7.9.5

